packages/vats/src/vat-board.js

```diff
@@ -1,7 +1,23 @@
-import { Far } from '@endo/far';
-import { makeBoard } from './lib-board.js';
+// @ts-check
+import { M, prepareExo } from '@agoric/vat-data';
+import { prepareBoardKit } from './lib-board.js';
 
-export function buildRootObject() {
-  const board = makeBoard();
-  return Far('board', { getBoard: () => board });
+const RootI = M.interface('VatBoard', {
+  getBoard: M.call().returns(M.remotable()),
+});
+
+/**
+ * @param {unknown} _vatPowers
+ * @param {unknown} _vatParameters
+ * @param {import('@agoric/vat-data').Baggage} baggage
+ */
+export function buildRootObject(_vatPowers, _vatParameters, baggage) {
+  const makeBoardKit = prepareBoardKit(baggage);
+  const { board } = makeBoardKit();
+  const root = prepareExo(baggage, 'vat-board', RootI, {
+    getBoard() {
+      return board;
+    },
+  });
+  return root;
```

Comment:
It seems unnecessary to store the root object in baggage; isn't it always reachable? Related: https://github.com/Agoric/agoric-sdk/pull/6916/files?short_path=0e323df#diff-0e323df3f817ef65ce44694815595144daed035213ba3b376f8716b45663b49a
```suggestion
  return Far('vat-board', { getBoard: () => board });
```

Reply:
I'm not sure what you mean by "always reachable". And I'm not sure which part of that diff is relevant.

If we don't store the root in baggage, it will get re-created with a new identity on upgrade, no? Are you saying clients should be prepared for that trauma? I suppose that's somewhat reasonable, but I don't know how we can know that in general.

Reply:
@dckc I think he's referring to https://github.com/Agoric/agoric-sdk/pull/6916/files#diff-0e323df3f817ef65ce44694815595144daed035213ba3b376f8716b45663b49aR64
> As a special case, the root object returned from v2's `buildRootObject()` is automatically associated with exportID `o+0` (see [How Liveslots Uses the Vatstore](https://github.com/Agoric/agoric-sdk/blob/master/packages/swingset-liveslots/src/vatstore-usage.md#counters)).

I take it that v1's root object had the same export so they always have the same identity.